woocommerce/woocommerce-ios
# Aztec editor for Product description: part 1 - no format toolbar

Aztec view controller - part 1 - without format bar for #1421 

- [x] ⚠️ update PR base after https://github.com/woocommerce/woocommerce-ios/pull/1430 is merged

## Changes
- Added Aztec editor pod `WordPress-Editor-iOS`
- Created `AztecEditorViewController` that contains the Aztec editor views (HTML/Rich Text views) and a placeholder without any formatting support yet (#1421 subtask: Aztec view controller - part 2 - format bar support)
  - Moved some configurable editor UI properties to `EditorViewProperties` (lemme know if there is a better naming!)
  - Extracted UI configurations to `AztecUIConfigurator`
  - Extracted `TextViewAttachmentDelegate` implementation to `AztecTextViewAttachmentHandler` without media support yet (a separate subtask in #1421)
- Created `EditorFactory` to instantiate a `Editor & UIViewController` for Product description
- In `ProductsViewController`, temporarily navigated to edit Product description screen on tapping a Product

## Testing
- Launch the app
- Go to the Products tab
- Tap on a Product --> try editing the product description, removing all the text to see the placeholder, and adding some text again

## Example screenshots

some text | empty text
-- | --
![Simulator Screen Shot - iPhone SE - 2019-11-06 at 17 27 37](https://user-images.githubusercontent.com/1945542/68286709-8006c380-00bc-11ea-9732-c7081c40b1ab.png) | ![Simulator Screen Shot - iPhone SE - 2019-11-06 at 17 27 42](https://user-images.githubusercontent.com/1945542/68286710-8006c380-00bc-11ea-8d27-e4b6ad91a4eb.png)


Update release notes:
